assignments/assignment7-prisma.md

```diff
@@ -22,16 +27,211 @@ Building on your existing Prisma application from Lesson 6b, you'll enhance it w
 
 ## Assignment Tasks
 
-### 1. Enhanced Task Analytics API
+### 1. Add Priority Field to Schema
 
-#### a. User Productivity Analytics Endpoint
-Create a new endpoint `GET /api/analytics/users/:id` that provides comprehensive user statistics:
+#### a. Update Prisma Schema
 
-**Requirements:**
-- Use `groupBy` to count tasks by completion status
-- Include recent task activity (last 10 tasks)
-- Calculate weekly progress metrics
-- Use eager loading to fetch related user data efficiently
+You need to add a `priority` field to your Task model. Open your `prisma/schema.prisma` file and update the Task model:
+
+```prisma
+model Task {
+  id           Int      @id @default(autoincrement())
+  title        String   @db.VarChar(255)
+  isCompleted Boolean  @default(false) @map("is_completed")
+  priority    String   @default("medium") @db.VarChar(10) // Add this line
+  userId      Int       @map("user_id")
+  createdAt   DateTime @default(now()) @db.Timestamp(6) @map("created_at")
+  User        User     @relation(fields: [userId], references: [id], onDelete: NoAction, onUpdate: NoAction)
+  @@unique([id, userId])
+  @@map("tasks")
+}
```

Comment:
Seems like we'd want to use a prisma enum for the low, medium, high, so as to permit only those values.